src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -0,0 +1,7757 @@
+//
+void CodeGen::genLockedInstructions(GenTreeOp* treeNode)
+{
+    NYI_RISCV64("genLockedInstructions-----unimplemented/unused on RISCV64 yet----");
+}
+
+//------------------------------------------------------------------------
+// genCodeForCmpXchg: Produce code for a GT_CMPXCHG node.
+//
+// Arguments:
+//    tree - the GT_CMPXCHG node
+//
+void CodeGen::genCodeForCmpXchg(GenTreeCmpXchg* treeNode)
+{
+    NYI_RISCV64("genCodeForCmpXchg-----unimplemented/unused on RISCV64 yet----");
+}
+
+static inline bool isImmed(GenTree* treeNode)
+{
+    assert(treeNode->OperIsBinary());
+
+    if (treeNode->gtGetOp2()->isContainedIntOrIImmed())
+    {
+        return true;
+    }
+
+    return false;
+}
+
+instruction CodeGen::genGetInsForOper(GenTree* treeNode)
+{
+    var_types  type = treeNode->TypeGet();
+    genTreeOps oper = treeNode->OperGet();
+    GenTree*   op1  = treeNode->gtGetOp1();
+    GenTree*   op2;
+    emitAttr   attr  = emitActualTypeSize(treeNode);
+    bool       isImm = false;
+
+    instruction ins = INS_ebreak;
+
+    if (varTypeIsFloating(type))
+    {
+        switch (oper)
+        {
+            case GT_ADD:
+                if (attr == EA_4BYTE)
+                {
+                    ins = INS_fadd_s;
+                }
+                else
+                {
+                    ins = INS_fadd_d;
+                }
+                break;
+            case GT_SUB:
+                if (attr == EA_4BYTE)
+                {
+                    ins = INS_fsub_s;
+                }
+                else
+                {
+                    ins = INS_fsub_d;
+                }
+                break;
+            case GT_MUL:
+                if (attr == EA_4BYTE)
+                {
+                    ins = INS_fmul_s;
+                }
+                else
+                {
+                    ins = INS_fmul_d;
+                }
+                break;
+            case GT_DIV:
+                if (attr == EA_4BYTE)
+                {
+                    ins = INS_fdiv_s;
+                }
+                else
+                {
+                    ins = INS_fdiv_d;
+                }
+                break;
+            case GT_NEG:
+                NYI_RISCV64("GT_NEG-----unimplemented/unused on RISCV64 yet----");
+                break;
+
+            default:
+                NYI_RISCV64("Unhandled oper in genGetInsForOper() - float");
+                unreached();
+                break;
+        }
+    }
+    else
+    {
+        switch (oper)
+        {
+            case GT_ADD:
+                isImm = isImmed(treeNode);
+                if (isImm)
+                {
+                    if ((attr == EA_8BYTE) || (attr == EA_BYREF))
+                    {
+                        ins = INS_addi;
+                    }
+                    else
+                    {
+                        assert(attr == EA_4BYTE);
+                        ins = INS_addi;
+                    }
```

Comment:
Is it intentional that both cases are `INS_addi`?

Reply:
Sorry. My mistake.